Renovate quality tools

akx opened the pull request:
This PR:

* removes old `isort` configuration that hasn't been in use since 99901896ccea794fd012b81fbf747c3476d98388
* replaces `black` formatting with `ruff format` (though `black` is still a hidden dependency via `hf-doc-builder`
* broadens quality controls to everywhere in the repo (since they're instant to run now)
* conservatively enables the Ruff `UP` (upgrade) rules and auto-fixes what can be auto-fixed.

This is similar to e.g. https://github.com/huggingface/transformers/pull/27144 and https://github.com/huggingface/diffusers/pull/5841.

BenjaminBossan reviewed (changes requested):
Thanks a lot for this update, which looks indeed very useful.

Doing a full review is a bit difficult because the diff is so large. Could we maybe add some exceptions to the ruff checks for the time being? Specifically, I'm thinking of:

1. https://docs.astral.sh/ruff/rules/redundant-open-modes/
2. https://docs.astral.sh/ruff/rules/utf8-encoding-declaration/
3. https://docs.astral.sh/ruff/rules/f-string/

This should take care of the majority of changes and thus lead to a much smaller diff and make the review easier. Then I can discuss with my co-maintainers if we want to have these exceptions or not, and if we don't, remove them in a later PR.

akx commented:
@BenjaminBossan Sure! I split the single `UP` autofix commit into multiple, that can be reviewed in GitHub one by one ✨

BenjaminBossan commented on src/peft/auto.py:
```diff
@@ -50,7 +49,7 @@ class _BaseAutoPeftModel:
     def __init__(self, *args, **kwargs):
         # For consistency with transformers: https://github.com/huggingface/transformers/blob/91d7df58b6537d385e90578dac40204cb550f706/src/transformers/models/auto/auto_factory.py#L400
-        raise EnvironmentError(
+        raise OSError(
```
As the comment above suggests, let's stick with `EnvironmentError`, even if it's not the correct error type here.

akx replied:
I believe the "consistency" there refers to raising an error in `__init__`, not the particular error type, but sure, added a commit to `noqa` that instead.

BenjaminBossan replied:
No, this refers to the type of error. I know because I flagged the error type initially :)

BenjaminBossan reviewed (commented):
Thanks for splitting up the commits. I'll have a discussion with the co-maintainers of the specific rules added here and we'll come back with a conclusion.

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/peft/pr_1421). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

BenjaminBossan commented on pyproject.toml:
```diff
@@ -1,33 +1,33 @@
+    "E501", # Line length (handled by ruff-format)
+    "E741", # Ambiguous variable name
+    "UP007", # X | Y style Unions
+    "W605", # Invalid escape sequence
```
What is the reasoning for ignoring this?

akx replied:
It was ignored in the preimage. I'll check if any of the original ignores can be safely removed.

akx replied:
Okay, E741 and W605 were easy enough to fix, so did that.

BenjaminBossan reviewed (commented):
Thanks a lot, the individual changes look sound and after discussing with the co-maintainers, we agree with the rules and exceptions chosen here.

Could you please check the two scripts that have yet to be reformatted according to CI? Thanks.

akx commented:
@BenjaminBossan Done, rebased, removed a couple of ignores by either autofixing or noqa'ing (there are 29 functions that are too complex according to default C901, so not touching those).

BenjaminBossan reviewed (approved):
Thanks a lot for moving the code quality checks over completely to ruff. LGTM.
